client/my-sites/comments/controller.js

```diff
@@ -12,27 +13,57 @@ import route from 'lib/route';
 import CommentsManagement from './main';
 import controller from 'my-sites/controller';
 
-export const comments = function( context ) {
+export const isStatusValid = status => {
+	const validStatuses = [ 'pending', 'approved', 'spam', 'trash', 'all' ];
+	return includes( validStatuses, status );
+};
+
+export const getRedirect = ( status, siteSlug ) => {
+	const statusValidity = isStatusValid( status );
+	if ( status === siteSlug ) {
+		return '/comments/pending/' + siteSlug;
+	}
+	if ( ! statusValidity && ! siteSlug ) {
+		return '/comments';
+	}
+	if ( ! statusValidity && siteSlug ) {
+		return '/comments/pending/' + siteSlug;
+	}
+	if ( statusValidity && ! siteSlug ) {
+		return '/comments/' + status;
+	}
+	return false;
```

Comment:
Maybe `null` would be a better default here? I tend to dislike returning mixed types from an undocumented function.